src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -679,6 +691,37 @@ private static Expression AddResponseWritingToMethodCall(Expression methodCall,
 
             if (feature?.CanHaveBody == true)
             {
+                if (isRawBodyType)
+                {
+                    var bodyReader = httpContext.Request.BodyReader;
+
+                    while (true)
+                    {
+                        var result = await bodyReader.ReadAsync();
+                        var buffer = result.Buffer;
+
+                        if (result.IsCompleted)
+                        {
+                            if (bodyType == typeof(ReadOnlySequence<byte>))
+                            {
+                                // REVIEW: Does this need to be a copy? We can tell users to consume the buffer
+                                // immediately in the action
+                                return (buffer, true);
+                            }
+
+                            // LOH be damned
+                            if (bodyType == typeof(ReadOnlyMemory<byte>))
+                            {
+                                return (new ReadOnlyMemory<byte>(buffer.ToArray()), true);
```

Comment:
Assuming that not copying is correct for the entire `ReadOnlySequence` above, then here you could check to see whether the ROS is a single segment and if it is then return `buffer.First`

Reply:
I'm not sure if we want to support this but I figured I'd put it in the PR and have a discussion. Copying is the safest thing to do here but I'm trying to figure out if there's a type we can use to indicate that a copy shouldn't be made (ReadOnlySequence\<byte\>).

Reply:
Decided to only support ReadOnlySequence\<byte\>